src/appengine/handlers/upload_testcase.py

```diff
@@ -609,25 +608,25 @@ def __init__(self, name, value):
     io.BytesIO.__init__(self, value)
 
 
-class UploadHandlerOAuth(base_handler.Handler, UploadHandlerCommon):
+class UploadHandlerOAuth(base_handler_flask.Handler, UploadHandlerCommon):
   """Handler that uploads the testcase file (OAuth)."""
 
   # pylint: disable=unused-argument
   def before_render_json(self, values, status):
     """Add upload info when the request fails."""
-    values['uploadUrl'] = self.request.host_url + UPLOAD_URL
+    values['uploadUrl'] = request.host_url + UPLOAD_URL
 
   def get_upload(self):
     """Get the upload."""
-    uploaded_file = self.request.POST.get('file')
+    uploaded_file = request.get('file')
```

Comment:
Actually, seems like I didn't test this properly last time. 

This needs to be updated:

```
-    uploaded_file = request.get('file')
-    if not isinstance(uploaded_file, cgi.FieldStorage):
+    uploaded_file = request.files.get('file')
+    if not uploaded_file:
       raise helpers.EarlyExitException('File upload not found.', 400)

-    bytes_io = NamedBytesIO(uploaded_file.filename, uploaded_file.file.read())
+    bytes_io = NamedBytesIO(uploaded_file.filename, uploaded_file.stream.read())
     key = blobs.write_blob(bytes_io)
     return blobs.get_blob_info(key)
```

Reply:
(Would be a good idea to check documentation before making changes in the future).